### ISSUE TITLE
left censoring vs truncation

### ISSUE THREAD

**Issue by Atan1988** [ChrisWaller26/brms_freq_sev#24]
Currently, we have been generally working under the framework of left truncation with regard to claim counts. When trying to apply this framework to a more practical problem, I realize that claims valued at or settled below deductible/SIR often times still get reported to the insurance company. In situations, where the actual number of claim counts are low (I am dealing with about 64 claims with observed dollars, but close to 200 claims overall), there would be an advantage to utilize the 0 dollar claims.

- This would mean that loss < ded wouldn't just drop out during the data generation process, but left censored to be a value of 0 net of deductible
- In practice, potentially not all claims are being reported if under deductible, and there might need to be a hidden truncation parameter ded2 = p * ded
- Not sure how does this work with XS insurance, where the underlying limit would be above 5M. 
- It would be generally interesting to see how does adding the left censored claims in impact the recovery of original parameters

**Comment by ChrisWaller26** [comment]
I agree that this is, in practice, often the case. The complication comes from estimating the value of p in this example. I think, for now at least, we should assume losses below deductibles go unreported for simplicity and then work out an approach once that all works to a satisfactory level.

One approach could be to model claim counts for zero claims separately from non-zero ones. For example:

fit_freq_zero = bf(claimcount0 ~ 1 + region) + poisson()
fit_freq = bf(claimcount ~ 1 + region) + poisson()

where claimcount0 is the number of claims with a zero loss.

This would allow us to retain the assumption that only losses above the deductible are reported and then study separately the distribution of zero-loss claims. There may be separate trends, e.g. companies which are more likely to report claims below deductible tend to be better organised/more risk averse.

**Comment by ChrisWaller26** [comment]
An alternative could be to add in a Bernoulli component to the model, such that a Bernoulli sample is taken first and this determines if the loss is zero or non-zero. If it is the latter, it then samples from the severity model. I think this would be more complicated using BRMs, though.
